osu.Game/Rulesets/Mods/Mod.cs

```diff
@@ -148,15 +149,39 @@ public void CopyFrom(Mod source)
             if (source.GetType() != GetType())
                 throw new ArgumentException($"Expected mod of type {GetType()}, got {source.GetType()}.", nameof(source));
 
-            foreach (var (_, prop) in this.GetSettingsSourceProperties())
+            foreach (var (_, property) in this.GetSettingsSourceProperties())
             {
-                var targetBindable = (IBindable)prop.GetValue(this)!;
-                var sourceBindable = (IBindable)prop.GetValue(source)!;
+                var targetBindable = (IBindable)property.GetValue(this)!;
+                var sourceBindable = (IBindable)property.GetValue(source)!;
 
                 CopyAdjustedSetting(targetBindable, sourceBindable);
             }
         }
 
+        /// <summary>
+        /// Copies all mod setting values sharing same <see cref="MemberInfo.Name"/> from <paramref name="source"/> into this instance.
+        /// </summary>
+        /// <param name="source">The mod to copy properties from.</param>
+        internal void CopySharedSettings(Mod source)
+        {
+            Dictionary<string, object> oldSettings = new Dictionary<string, object>();
+
+            foreach (var (_, property) in source.GetSettingsSourceProperties())
+            {
+                oldSettings.Add(property.Name.ToSnakeCase(), property.GetValue(source)!);
+            }
+
+            foreach (var (_, property) in this.GetSettingsSourceProperties())
+            {
+                var targetBindable = (IBindable)property.GetValue(this)!;
+
+                if (!oldSettings.TryGetValue(property.Name.ToSnakeCase(), out object? sourceSetting))
+                    continue;
+
+                CopyAdjustedSetting(targetBindable, sourceSetting);
```

Comment:
- This should probably be checking that both bindables have the same inner data type to avoid a crash (with test coverage preferably please, since I'm not sure we have an existing case).
- And even if they're the same type, I'm not really sure about this in general. Take flashlight, for instance. Currently [all](https://github.com/ppy/osu/blob/d679703fa24dcf8ed12f95febd0eeb962e3552a6/osu.Game.Rulesets.Osu/Mods/OsuModFlashlight.cs#L35-L40) [four](https://github.com/ppy/osu/blob/d679703fa24dcf8ed12f95febd0eeb962e3552a6/osu.Game.Rulesets.Catch/Mods/CatchModFlashlight.cs#L18-L23) [default](https://github.com/ppy/osu/blob/d679703fa24dcf8ed12f95febd0eeb962e3552a6/osu.Game.Rulesets.Mania/Mods/ManiaModFlashlight.cs#L19-L24) [rulesets](https://github.com/ppy/osu/blob/d679703fa24dcf8ed12f95febd0eeb962e3552a6/osu.Game.Rulesets.Taiko/Mods/TaikoModFlashlight.cs#L19-L24) use slightly different values of `SizeMultiplier`. Does transferring this value on ruleset change make sense if they're seemingly different quantities? I dunno.

Reply:
> This should probably be checking that both bindables have the same inner data type to avoid a crash (with test coverage preferably please, since I'm not sure we have an existing case).

Done.

> And even if they're the same type, I'm not really sure about this in general. Take flashlight, for instance. Currently all four default rulesets use slightly different values of SizeMultiplier. Does transferring this value on ruleset change make sense if they're seemingly different quantities? I dunno.

Currently converts the value to the closest relative value with respect to `MinValue` and `MaxValue`.